Add support for generating more pair consumers

weinig opened the pull request:
#### 21915b3edbe66caaaaac2b821dd67504a293081c
<pre>
Add support for generating more pair consumers
<a href="https://bugs.webkit.org/show_bug.cgi?id=289432">https://bugs.webkit.org/show_bug.cgi?id=289432</a>

Reviewed by Tim Nguyen.

Switch more consumers that use the form `[...]{1,2}` in their
grammars to match the modeling used already where a coalescing
CSSValuePair is used. With the consumer generated, the style
builder and computed style extractors need to be updated to
match.

These allow generation of:
    -webkit-initial-letter
    scroll-snap-align

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Inline.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Inline.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ScrollSnap.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ScrollSnap.h:
* Source/WebCore/style/StyleBuilderConverter.h:

Canonical link: <a href="https://commits.webkit.org/291873@main">https://commits.webkit.org/291873@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0a0b0cf25116ec7c65d6f166759482de6007d5c0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94274 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/13861 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/3631 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99286 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/44802 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96324 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14161 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22291 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/71922 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29262 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97276 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/10519 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85127 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52268 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10208 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/2811 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44120 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/80440 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/2902 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101330 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21326 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/15542 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/80926 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/21578 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81140 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80308 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20018 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/24859 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2231 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/14544 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21310 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/26489 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/20997 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/24457 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/22738 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b4be6aea74964daba7f420cbe62833f47218cc14. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/42159

nt1m reviewed (approved):
r=me pending green EWS

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0a0b0cf25116ec7c65d6f166759482de6007d5c0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/42159

weinig committed:
Add support for generating more pair consumers
https://bugs.webkit.org/show_bug.cgi?id=289432

Reviewed by Tim Nguyen.

Switch more consumers that use the form `[...]{1,2}` in their
grammars to match the modeling used already where a coalescing
CSSValuePair is used. With the consumer generated, the style
builder and computed style extractors need to be updated to
match.

These allow generation of:
    -webkit-initial-letter
    scroll-snap-align

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Inline.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+Inline.h:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ScrollSnap.cpp:
* Source/WebCore/css/parser/CSSPropertyParserConsumer+ScrollSnap.h:
* Source/WebCore/style/StyleBuilderConverter.h:

Canonical link: https://commits.webkit.org/291873@main

webkit-commit-queue commented:
Committed 291873@main (21915b3edbe6): <https://commits.webkit.org/291873@main>

Reviewed commits have been landed. Closing PR #42159 and removing active labels.
